src/vxworks/mod.rs

```diff
@@ -463,6 +474,46 @@ cfg_if! {
                     .finish()
             }
         }
+
+        impl PartialEq for sa_u_t {
+            fn eq(&self, other: &sa_u_t) -> bool {
+                unsafe { self.sa_handler == other.sa_handler }
+            }
+        }
+        impl Eq for sa_u_t {}
+        impl ::fmt::Debug for sa_u_t {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                f.debug_struct("sa_u_t")
+                    .field("sa_handler", unsafe { &self.sa_handler })
+                    .field("sa_sigaction", unsafe { &self.sa_sigaction })
+                    .finish()
+            }
+        }
+        impl ::hash::Hash for sa_u_t {
+            fn hash<H: ::hash::Hasher>(&self, state: &mut H) {
+                unsafe { self.sa_handler.hash(state) };
+            }
+        }
+
+        impl PartialEq for sigval {
+            fn eq(&self, other: &sigval) -> bool {
+                unsafe { self.sival_int == other.sival_int }
```

Comment:
Should be sival_ptr. Otherwise two different values that compare "equal" can produce different hashes (line 514)!